core/src/types/operator/operator_futures.rs

```diff
@@ -0,0 +1,137 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//! Futures provides the futures generated by [`Operator`]
+//!
+//! By using futures, users can add more options for operation.
+
+use std::mem;
+use std::pin::Pin;
+use std::task::{Context, Poll};
+
+use futures::future::BoxFuture;
+use futures::Future;
+use futures::FutureExt;
+
+use crate::ops::*;
+use crate::raw::*;
+use crate::*;
+
+/// OperatorFuture is the future generated by [`Operator`].
+///
+/// The future will consume all the input to generate a future.
+pub(crate) enum OperatorFuture<T, F> {
+    /// Idle state, waiting for the future to be polled
+    Idle(
+        /// The accessor to the underlying object storage
+        FusedAccessor,
+        /// The path of string
+        String,
+        /// The input args
+        T,
+        /// The function which will move all the args and return a static future
+        fn(FusedAccessor, String, T) -> BoxFuture<'static, Result<F>>,
```

Comment:
Can the lifetime be narrowed down here?

Reply:
We don't need lifetime here: all value are moved, there is no reference.

Reply:
Ok.